src/scene/graph-node.js

```diff
@@ -534,40 +555,13 @@ class GraphNode extends EventHandler {
      * const entities = parent.find('name', 'Test');
      */
     find(attr, value) {
-        let result, results = [];
-        const len = this._children.length;
-
-        if (attr instanceof Function) {
-            const fn = attr;
-
-            result = fn(this);
-            if (result)
-                results.push(this);
-
-            for (let i = 0; i < len; i++) {
-                const descendants = this._children[i].find(fn);
-                if (descendants.length)
-                    results = results.concat(descendants);
-            }
-        } else {
-            let testValue;
-
-            if (this[attr]) {
-                if (this[attr] instanceof Function) {
-                    testValue = this[attr]();
-                } else {
-                    testValue = this[attr];
-                }
-                if (testValue === value)
-                    results.push(this);
-            }
+        const results = [];
+        const test = createTest(attr, value);
 
-            for (let i = 0; i < len; ++i) {
-                const descendants = this._children[i].find(attr, value);
-                if (descendants.length)
-                    results = results.concat(descendants);
-            }
-        }
+        this.forEach((node) => {
+            if (test(node))
+                results.push(node);
```

Comment:
shouldn't this recurse into descendants?

Reply:
check `this.forEach` implementation